python/artm/master_component.py

```diff
@@ -631,3 +690,113 @@ def get_info(self):
+           - update_after(list of int): number of batches to be passed for Phi synchronizations
+           - apply_weight(list of float): weight of applying new counters
+             (len == len of update_after)
+           - decay_weight(list of float): weight of applying old counters
+             (len == len of update_after)
+           - async(bool): use or not the async implementation of the EM-algorithm
+        """
+        args = messages.FitOnlineMasterModelArgs()
+        if batch_filenames is not None:
+            args.ClearField('batch_filename')
+            for filename in batch_filenames:
+                args.batch_filename.append(filename)
+
+        if batch_weights is not None:
+            args.ClearField('batch_weight')
+            for weight in batch_weights:
+                args.batch_weight.append(weight)
+
+        if update_after is not None:
+            args.ClearField('update_after')
+            for value in update_after:
+                args.update_after.append(value)
+
+        if update_after is not None:
+            args.ClearField('update_after')
+            for value in update_after:
+                args.update_after.append(value)
+
+        if apply_weight is not None:
+            args.ClearField('apply_weight')
+            for value in apply_weight:
+                args.apply_weight.append(value)
+
+        if decay_weight is not None:
+            args.ClearField('decay_weight')
+            for value in decay_weight:
+                args.decay_weight.append(value)
+
+        if async is not None:
+            args.async = async
+
+        self._lib.ArtmFitOnlineMasterModel(self.master_id, args)
+
+    def transform(self, batches=None, batch_filenames=None,
+                  theta_matrix_type=None, predict_class_id=None):
+        """Args:
+           - batches(list of batches): list of instances of Batch
+           - batch_weights(list of float): weights of batches to transform
+           - theta_matrix_type(int): type of matrix to be returned
+           - predict_class_id(int): type of matrix to be returned
+           Returns:
+           - messages.ThetaMatrix object
+        """
+        args = messages.TransformMasterModelArgs()
+        if batches is not None:
+            args.ClearField('batch')
+            for batch in batches:
+                batch_ref = args.batch.add()
+                batch_ref.CopyFrom(batch)
+
+        if batch_filenames is not None:
+            args.ClearField('batch_filename')
+            for filename in batch_filenames:
+                args.batch_filename.append(filename)
+
+        if theta_matrix_type is not None:
+            args.theta_matrix_type = theta_matrix_type
+
+        if predict_class_id is not None:
+            args.predict_class_id = predict_class_id
+
+        result = self._lib.ArtmRequestTransformMasterModel(self.master_id, args)
+
+        theta_matrix = messages.ThetaMatrix()
+        theta_matrix.ParseFromString(result)
```

Comment:
Just curious - why `wrapper.py` doesn't do `theta_matrix = messages.ThetaMatrix()` and `theta_matrix.ParseFromString(result)`? You have to manually do it every time after calling `ArtmRequestXxxx`. Not that it is related to this request, but it would be nice to fix this (unless I'm missing something)